gc.c

```diff
@@ -11822,7 +11830,10 @@ ruby_gc_set_params(void)
 
     /* RUBY_GC_HEAP_INIT_SLOTS */
     if (get_envparam_size("RUBY_GC_HEAP_INIT_SLOTS", &gc_params.heap_init_slots, 0)) {
-        gc_set_initial_pages(objspace);
+        gc_set_initial_pages(objspace, TRUE);
+    }
+    else {
+        gc_set_initial_pages(objspace, FALSE);
```

Comment:
I think this resets all of the `heap_allocatable_pages`.

This branch:

```
$ ruby --disable-gems -e "GC.stat_heap.each { |k, v| puts k.to_s + '=' + v[:heap_allocatable_pages].to_s }"
0=0
1=0
2=0
3=0
4=0
```

Master branch:

```
$ ruby --disable-gems -e "GC.stat_heap.each { |k, v| puts k.to_s + '=' + v[:heap_allocatable_pages].to_s }"
1=11
2=22
3=48
4=97
```

Reply:
Good catch, I fixed it.